lib/schedule/swing.ex

```diff
@@ -116,4 +116,28 @@ defmodule Schedule.Swing do
     do: nil
 
   defp trip_or_trip_id_to_trip(trip_id, trips_by_id), do: Map.fetch!(trips_by_id, trip_id)
+
+  @spec first_trip_from_piece(Schedule.Piece.t(), Trip.by_id()) ::
+          Trip.id() | Schedule.Trip.t() | Schedule.AsDirected.t()
+  defp first_trip_from_piece(piece, trips_by_id) do
+    first_full_trip =
+      piece.trips
+      |> Stream.map(&trip_or_trip_id_to_trip(&1, trips_by_id))
+      |> Stream.filter(fn trip -> !is_nil(trip) end)
+      |> Enum.take(1)
+      |> List.first()
+
+    case {first_full_trip, piece.start_mid_route?} do
+      {nil, %{trip: mid_route_trip}} ->
+        mid_route_trip |> trip_or_trip_id_to_trip(trips_by_id)
```

Comment:
**question:** is this one of the data edge cases you were referring to? If I understand correctly, this is the case in which the piece starts with a swing but has no other trips. I'd say to make the same update you do in the other case to make sure the `run_id` is correct juuuust in case this ever comes up, but I doubt it's likely to ever happen.

Reply:
Yeah this is that edge case! In this case though, what would the correct `run_id` be? Since there are no other full trips, would it be just the `run_id` of the swing?

Reply:
I was thinking you could use the `run_id` from the `piece`, provided that that's actually populated and correct.

Reply:
Ah! Do you have a sense of how reliable that field is? Wondering if that could be used for the `run_id` so we no longer have to look for `first_full_trip` to get the run_id when there is a mid route swing-on

Reply:
I think it should be pretty reliable. If I'm not mistaken that value would get populated [here](https://github.com/mbta/skate/blob/a837fa82f60652464deeae640c8152b04cf6bdff/lib/schedule/hastus/activity.ex#L135), and it appears that it's ultimately using the value of `run_id` from the relevant row in `activities.csv`.